### PR TITLE
Phase 6: Model Scale-Up — Deeper/Wider Transolver Backbone

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis

The current 3-layer, 96-slice Transolver uses only ~38GB of the available 96GB VRAM. We have never tested whether the backbone is capacity-limited. After exhausting regularization, optimizer, and output-head improvements, the next logical question is: **does more representational capacity improve surface MAE?**

**Why this might help:** p_tan=29.1 (tandem transfer) is 2.4× worse than p_in=12.2, suggesting the model struggles with complex cross-foil interactions. Deeper or wider models have more capacity to represent these interactions. We have 2.5× unused VRAM headroom — this is a basic scaling experiment we haven't run.

**Mechanism:** More Transolver blocks → deeper feature hierarchies, richer attention over physics slices. More attention slices → finer-grained partitioning of the flow field into physics-consistent clusters.

## Instructions

The baseline command uses `--n_layers 3 --slice_num 96`. You only need to vary these two flags — no other code changes required.

**Full baseline command (all runs use this base):**
```bash
python train.py --agent frieren --wandb_group phase6/model-scale-up \
  --asinh_pressure --asinh_scale 0.75 --field_decoder --adaln_output --use_lion --lr 2e-4 \
  --aug aoa_perturb --aug_full_dsdf_rot --high_p_clamp \
  --tandem_ramp --domain_layernorm --domain_velhead --ema_decay 0.999 --weight_decay 5e-5 \
  --cosine_T_max 160 --disable_pcgrad --pressure_first --pressure_deep \
  --residual_prediction --surface_refine --surface_refine_hidden 192 --surface_refine_layers 3
```

| GPU | wandb_name | Extra flags | Seed |
|-----|-----------|-------------|------|
| 0 | frieren/baseline-3L-96s-s42 | --n_layers 3 --slice_num 96 | 42 |
| 1 | frieren/baseline-3L-96s-s73 | --n_layers 3 --slice_num 96 | 73 |
| 2 | frieren/deeper-5L-96s-s42 | --n_layers 5 --slice_num 96 | 42 |
| 3 | frieren/deeper-5L-96s-s73 | --n_layers 5 --slice_num 96 | 73 |
| 4 | frieren/wider-3L-160s-s42 | --n_layers 3 --slice_num 160 | 42 |
| 5 | frieren/wider-3L-160s-s73 | --n_layers 3 --slice_num 160 | 73 |
| 6 | frieren/deep-wide-4L-128s-s42 | --n_layers 4 --slice_num 128 | 42 |
| 7 | frieren/deep-wide-4L-128s-s73 | --n_layers 4 --slice_num 128 | 73 |

**VRAM check:** After launching, verify memory usage on each GPU (`nvidia-smi`). If any run OOMs, reduce that config's `--slice_num` by 16 or `--n_layers` by 1 and relaunch. Expected VRAM: 3L/96s ~38GB, 5L/96s ~55GB, 3L/160s ~55GB, 4L/128s ~58GB — all should fit in 96GB.

**Epoch count note:** Larger models have slower epochs and will complete fewer epochs within the 180-min timeout. This is expected. If larger models show lower val/loss at their final epoch (even with fewer total epochs), that's a win.

**What to report:**
- Surface MAE (p_in, p_oodc, p_tan, p_re) for each config
- Memory usage per GPU (GB)
- Number of epochs completed within timeout
- Whether larger models were still improving at timeout (check val/loss curve trend)

## Baseline

Current best (8-seed ensemble, PR #2080):
| p_in | p_oodc | p_tan | p_re |
|------|--------|-------|------|
| **12.2** | **6.7** | **29.1** | **5.8** |

Single-model 8-seed mean (target for single-model improvement):
| p_in | p_oodc | p_tan | p_re |
|------|--------|-------|------|
| 13.03 ± 0.39 | 7.83 ± 0.19 | 30.29 ± 0.47 | 6.45 ± 0.05 |

**W&B group:** `phase6/model-scale-up`

**Reproduce baseline:**
```bash
python train.py --agent frieren --wandb_group phase6/model-scale-up \
  --asinh_pressure --asinh_scale 0.75 --field_decoder --adaln_output --use_lion --lr 2e-4 \
  --aug aoa_perturb --aug_full_dsdf_rot --high_p_clamp --n_layers 3 --slice_num 96 \
  --tandem_ramp --domain_layernorm --domain_velhead --ema_decay 0.999 --weight_decay 5e-5 \
  --cosine_T_max 160 --disable_pcgrad --pressure_first --pressure_deep \
  --residual_prediction --surface_refine --surface_refine_hidden 192 --surface_refine_layers 3 \
  --seed 42 --wandb_name "frieren/baseline-3L-96s-s42"
```

### CODE REVIEW

**Comment by tcapelle** [comment]
STUDENT frieren:

## Results

All 8 runs completed (~180 min each). No code changes needed — only varied `--n_layers` and `--slice_num`.

### Metrics (Surface MAE Pressure — Best Checkpoint)

| Config | Seed | p_in | p_oodc | p_tan | p_re | val/loss | Best Epoch | Total Epochs | Epoch Time | VRAM |
|--------|------|------|--------|-------|------|----------|------------|--------------|------------|------|
| 3L/96s (baseline) | 42 | **13.0** | **7.6** | 31.1 | **6.4** | **0.3915** | 157 | 157 | 68s | 38.0GB |
| 3L/96s (baseline) | 73 | **12.9** | 7.9 | **30.3** | **6.4** | **0.3886** | 157 | 157 | 68s | 38.0GB |
| **3L/96s mean** | — | **12.9** | **7.8** | **30.7** | **6.4** | **0.3901** | — | **157** | **68s** | **38.0GB** |
| 5L/96s (deeper) | 42 | 19.0 | 10.4 | 31.5 | 8.6 | 0.4562 | 105 | 105 | 102s | 55.4GB |
| 5L/96s (deeper) | 73 | 16.3 | 10.0 | 30.4 | 8.2 | 0.4349 | 106 | 106 | 101s | 55.3GB |
| **5L/96s mean** | — | **17.6** | **10.2** | **30.9** | **8.4** | **0.4456** | — | **106** | **101s** | **55.4GB** |
| 3L/160s (wider) | 42 | 13.7 | 7.9 | 30.8 | 6.5 | 0.3959 | 141 | 141 | 76s | 40.5GB |
| 3L/160s (wider) | 73 | 13.0 | 8.3 | **30.2** | 6.5 | 0.3905 | 142 | 142 | 75s | 40.5GB |
| **3L/160s mean** | — | **13.3** | **8.1** | **30.5** | **6.5** | **0.3932** | — | **142** | **76s** | **40.5GB** |
| 4L/128s (deep-wide) | 42 | 15.9 | 9.4 | 32.0 | 7.5 | 0.4255 | 119 | 119 | 89s | 48.4GB |
| 4L/128s (deep-wide) | 73 | 15.6 | 9.7 | 31.5 | 7.8 | 0.4266 | 118 | 118 | 90s | 48.4GB |
| **4L/128s mean** | — | **15.8** | **9.6** | **31.8** | **7.7** | **0.4260** | — | **118** | **90s** | **48.4GB** |

### Delta vs Baseline (2-seed means)

| Config | p_in | p_oodc | p_tan | p_re |
|--------|------|--------|-------|------|
| 5L/96s (deeper) | **+36.3%** | **+31.6%** | +0.8% | **+31.2%** |
| 3L/160s (wider) | +3.1% | +4.5% | -0.7% | +1.6% |
| 4L/128s (deep-wide) | **+21.6%** | **+23.2%** | +3.4% | **+19.5%** |

### Were Models Still Improving at Timeout?

| Config | Final val_in_dist | Trend (last 5 epochs) | Status |
|--------|------------------|-----------------------|--------|
| 3L/96s | 0.265 | Flat (converged) | **Converged** |
| 5L/96s | 0.34-0.38 | Noisy but declining | **Still improving** — not close to convergence |
| 3L/160s | 0.272→0.260 | Steadily declining | **Still improving** — approaching baseline (0.265) |
| 4L/128s | 0.31-0.33 | Declining | **Still improving** — not close to convergence |

### W&B Run IDs

| Config | Run ID |
|--------|--------|
| baseline-3L-96s-s42 | u00xzh8z |
| baseline-3L-96s-s73 | 3y9kazhf |
| deeper-5L-96s-s42 | z8xu1h7q |
| deeper-5L-96s-s73 | u1zdhi63 |
| wider-3L-160s-s42 | mkxyc0mf |
| wider-3L-160s-s73 | fzabfqmz |
| deep-wide-4L-128s-s42 | i4o4g2qm |
| deep-wide-4L-128s-s73 | 910al9e9 |

W&B group: `phase6/model-scale-up`

### Commands Used

```bash
python train.py --agent frieren --wandb_group phase6/model-scale-up \
  --asinh_pressure --asinh_scale 0.75 --field_decoder --adaln_output --use_lion --lr 2e-4 \
  --aug aoa_perturb --aug_full_dsdf_rot --high_p_clamp \
  --n_layers {3,5,4} --slice_num {96,160,128} \
  --tandem_ramp --domain_layernorm --domain_velhead --ema_decay 0.999 --weight_decay 5e-5 \
  --cosine_T_max 160 --disable_pcgrad --pressure_first --pressure_deep \
  --residual_prediction --surface_refine --surface_refine_hidden 192 --surface_refine_layers 3 \
  --seed {42,73} --wandb_name "frieren/<config>"
```

### Analysis

**The wider model (3L/160s) is the most promising configuration.** At the time of timeout:

1. **3L/160s (wider)** nearly matches the baseline despite 15 fewer epochs (142 vs 157). The val_in_dist was still declining at 0.260 (vs baseline's converged 0.265). Given more training time, this configuration may match or slightly beat the baseline. The cost is modest: +2.5GB VRAM, +8s/epoch.

2. **5L/96s (deeper)** is significantly worse (+36% p_in, +31% p_oodc) and was still improving at timeout. However, it only trained for 106 epochs (~67% of baseline's epochs). The 5L model's per-epoch cost (102s vs 68s) means it needs 50% more wall-clock time to match baseline epoch count. It's unclear if convergence would bridge the gap.

3. **4L/128s (deep-wide)** splits the difference but is worse than both pure approaches. It needs more training time (118 epochs, 90s/epoch) and currently shows +22% p_in, +23% p_oodc.

4. Notably, **p_tan is similar across all configs** (~30-32). The tandem transfer metric doesn't benefit from extra capacity, suggesting it's limited by something other than model size (perhaps data distribution or the transfer gap itself).

### Suggested Follow-ups

1. **Run 3L/160s with longer training**: Use `--cosine_T_max 200` or increase MAX_TIMEOUT to give the wider model full convergence time. It was clearly still improving.
2. **Try 3L/128s**: A more modest width increase (128 vs 160 slices) would add less per-epoch overhead while testing if moderate width helps.
3. **Deeper models need proportionally longer schedules**: If revisiting 5L, pair it with `--cosine_T_max 240+` to account for the slower epoch time. The current 160-epoch cosine schedule is too aggressive for a model that can only train 106 epochs.

**Comment by tcapelle** [comment]
ADVISOR: Closing PR #2100 because Closing — capacity is not the bottleneck. No configuration beats the 3L/96s baseline:

| Config | p_in (Δ) | p_oodc (Δ) | p_tan (Δ) | p_re (Δ) |
|--------|----------|-----------|----------|---------|
| 3L/96s (baseline) | 12.9 | 7.8 | 30.7 | 6.4 |
| 5L/96s | 17.6 (+36%) | 10.2 (+31%) | 30.9 (+0.8%) | 8.4 (+31%) |
| 3L/160s | 13.3 (+3%) | 8.1 (+4.5%) | 30.5 (-0.7%) | 6.5 (+1.6%) |
| 4L/128s | 15.8 (+22%) | 9.6 (+23%) | 31.8 (+3.4%) | 7.7 (+20%) |

Critical insight: **p_tan is remarkably similar across all configs (~30-32).** This confirms the tandem transfer problem is NOT a capacity issue — it's a distribution/representation problem. The 3L/160s wider model was still improving at timeout but its metrics already suggest convergence toward baseline, not improvement beyond it. Well-executed experiment with a clear and valuable finding..
